src/gwt/src/org/rstudio/studio/client/workbench/codesearch/CodeSearchOracle.java

```diff
@@ -192,51 +192,41 @@ public void requestSuggestions(final Request request,
       // failed to short-circuit via the cache, hit the server
       codeSearch_.enqueRequest(request, callback); 
    }
-     
+   
+   // this method converts queries of the form 'foo:<line>:<col>' to
+   // an appropriate navigation target, transforming the query and
+   // file position as appropriate
    public CodeNavigationTarget navigationTarget(String query,
                                                 Suggestion suggestion)
    {
       CodeSearchSuggestion codeSearchSuggestion = (CodeSearchSuggestion) suggestion;
+      CodeNavigationTarget target = codeSearchSuggestion.getNavigationTarget();
       
-      // Allow queries of the form e.g. 'foo:15' to go to line '15' of a file.
-      // We parse the integer following the ':' if possible.
-      FilePosition filePos = codeSearchSuggestion.getNavigationTarget().getPosition();
-      if (codeSearchSuggestion.isFileTarget())
-      {
-         int colonIndex = query.indexOf(":");
-         if (colonIndex > 0)
-         {
-            String[] splat = query.split(":");
-            if (splat.length > 1)
-            {
-               int rowToNavigateTo = 0;
-               try
-               {
-                  rowToNavigateTo = Integer.parseInt(splat[1]);
-               }
-               catch (Exception e)
-               {}
-               
-               int colToNavigateTo = 0;
-               if (splat.length > 2)
-               {
-                  try
-                  {
-                     colToNavigateTo = Integer.parseInt(splat[2]);
-                  }
-                  catch (Exception e)
-                  {}
-               }
-               filePos = FilePosition.create(rowToNavigateTo, colToNavigateTo);
-            }
-            
-         }
-      }
+      // nothing to do for non-file targets
+      if (!codeSearchSuggestion.isFileTarget())
+         return target;
+      
+      // nothing to do if the query doesn't have ':'
+      int colonIndex = query.indexOf(":");
```

Comment:
Perhaps it's accounted for elsewhere, but do we want to have `fig:`, `tab:`, etc. filter on xref type?

Reply:
That would be nice but it's not very discoverable, so hard to know if people would ever use it.

Reply:
That's true. Probably not worth it then.